File: python/stack.py
class Node:
    
    def __init__(self, d):
        self.data = d
        self.prev = None


class Stack:
    def __init__(self):
        self.head = Node

    def peek(self):
        return self.head.data
    
    def push(self, item):
        new_node = Node(d=item)
        if self.head:
            new_node.prev = self.head
            self.head = new_node
            return self.head.data
        self.head = new_node

    def pop(self):
        if self.head:
            head = self.head
            self.head = head.prev
            return head.data
        return None

File: java/Stack.java
package java;

class Node{
    int data;
    Node prev;

    Node(int d){
        data = d;
        prev = null;
    }
}

class Stack{
    Node head;
    Stack(Node h){
        head = h;
    }

    private int peek(){
        return this.head.data;
    }

    private int push(Node item){
        Node new_node = new Node(item.data);
        if(this.head != null){
            new_node.prev = this.head;
            this.head = new_node;
            return this.head.data;
        }
        this.head = new_node;
        return this.head.data;
    }

    private int pop(){
        if(this.head != null){
            Node head = new Node(this.head.data);
            this.head = head.prev;
            return head.data;
        }
        return -1;
    }
}